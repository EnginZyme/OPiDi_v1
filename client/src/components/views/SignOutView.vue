<template>
  <div class="sign_out_view">
    <div class="sign_out_view__group">
      <UserAvatarImage size="large" />
    </div>
    <div>
      <p class="sign_out_view__name_text">{{ userName }}</p>
      <p class="sign_out_view__email_text">{{ userEmail }}</p>
    </div>
    <div class="sign_out_view__group">
      <Button
        label="Sign Out"
        class="base_button medium_button_"
        @click="signOut"
      />
    </div>
  </div>
</template>

<script>
import UserAvatarImage from "../base/UserAvatarImage";
import { mapGetters } from "vuex";
import types from "../../types";

/**
 * A component that renders the sign-out view.
 */
export default {
  name: "SignOutView",
  components: {
    UserAvatarImage,
  },
  methods: {
    /**
     * Signs the current user out.
     */
    signOut() {
      let auth2 = window.gapi.auth2.getAuthInstance();
      auth2.signOut().then(function() {});

      setTimeout(() => {
        this.$store.dispatch(types.DEREGISTER_USER).then(() => {
          // Redirect to "Authentication"
          this.$router.push({
            name: "Authentication",
          });
        });
      }, 1000);
    },
  },
  computed: {
    ...mapGetters(["userName", "userEmail"]),
  },
};
</script>

<style scoped>
.sign_out_view {
  display: grid;
  justify-content: center;
  padding-top: 8px;
  padding-bottom: 12px;
  width: 100%;
  height: 100%;
}
.sign_out_view__name_text {
  text-align: center;
  font-size: 18px;
  color: #000000;
}
.sign_out_view__email_text {
  text-align: center;
  font-size: 16px;
  color: #767676;
}
.sign_out_view__group {
  display: grid;
  justify-content: center;
}
</style>
