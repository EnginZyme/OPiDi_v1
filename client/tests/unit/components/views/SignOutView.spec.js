import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import SignOutView from "@/components/views/SignOutView";

const localVue = createLocalVue();

localVue.use(Vuex);

const sampleUserEmail = "bar@acme.com";
const sampleUserName = "bar baz";

describe("shallowMounted SignOutView", () => {
  let getters;
  let store;
  let wrapper;

  beforeEach(() => {
    getters = {
      userEmail: () => sampleUserEmail,
      userName: () => sampleUserName,
    };

    store = new Vuex.Store({
      getters,
    });

    wrapper = shallowMount(SignOutView, {
      store,
      localVue,
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
