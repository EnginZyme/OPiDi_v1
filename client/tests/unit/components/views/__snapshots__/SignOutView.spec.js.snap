// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted SignOutView renders correctly 1`] = `
<div
  class="sign_out_view"
>
  <div
    class="sign_out_view__group"
  >
    <useravatarimage-stub
      size="large"
    />
  </div>
   
  <div>
    <p
      class="sign_out_view__name_text"
    >
      bar baz
    </p>
     
    <p
      class="sign_out_view__email_text"
    >
      bar@acme.com
    </p>
  </div>
   
  <div
    class="sign_out_view__group"
  >
    <button
      class="base_button medium_button_"
      label="Sign Out"
    />
  </div>
</div>
`;
